src/kernel/src/fs/null/vnode.rs

```diff
@@ -33,6 +32,36 @@ fn access(vn: &Arc<Vnode>, _td: Option<&VThread>, access: Access) -> Result<(),
     todo!();
 }
 
+fn getattr(vn: &Arc<Vnode>) -> Result<VnodeAttrs, Box<dyn Errno>> {
+    todo!()
+}
+
+fn lookup(vn: &Arc<Vnode>, td: Option<&VThread>, name: &str) -> Result<Arc<Vnode>, Box<dyn Errno>> {
+    let null_mount: &NullNode = vn.data().downcast_ref().unwrap();
+
+    let lower = null_mount
+        .lower()
+        .lookup(td, name)
+        .map_err(LookupError::LookupFailed)?;
+
+    let vnode = if Arc::ptr_eq(&lower, vn) {
```

Comment:
Is this even possible? The `vn` here was produced by nullfs and the lower one we got here should always coming from elsewhere.

Reply:
Not sure